docs/en/sql-reference/functions/other-functions.md

```diff
@@ -2609,3 +2609,103 @@ Result:
 │                      286 │
 └──────────────────────────┘
 ```
+
+## generateRandomStructure
+
+Generates random table structure in a format `column1_name column1_type, column2_name column2_type, ...`.
+
+**Syntax**
+
+``` sql
+generateRandomStructure([number_of_columns, seed, allow_big_numbers, allow_enums, allow_decimals, allow_ip, allow_only_string_map_keys])
+```
+
+**Arguments**
+
+- `number_of_columns` — The desired number of columns in the result table structure. If set to 0 or `Null`, the number of columns will be random from 1 to 128. Default value: `Null`.
+- `seed` - Random seed to produce stable results. If seed is not specified or set to `Null`, it is randomly generated.
+- `allow_big_numbers` - Indicates if big number types (`Int128/UInt128/Int256/UInt256/Decimal128/Decinal256`) can be generated. Default value: true.
+- `allow_enums` - Indicates if enum types (`Enum8/Enum16`) can be generated. Default - true.
+- `allow_decimals` - Indicates if decimal types (`Decimal(P, S)`) can be generated. Default - true.
+- `allow_ip` - Indicates if ip types (`IPv4/IPv6`) can be generated. Default - true.
+- `allow_only_string_map_keys` - Indicates if Map key type can be only `String/FixedString`. Default - false.
```

Comment:
I need these flags because I am going to use this function in formats fuzzer and some binary formats don't support some types.